packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -13,49 +13,23 @@ import {
 import { Icon, IIconProps } from '../../Icon';
 import { DirectionalHint } from '../../common/DirectionalHint';
 import { ContextualMenu, IContextualMenuProps } from '../../ContextualMenu';
-import { IButtonProps, IButton } from './Button.Props';
+import { IButtonProps, IButton, IButtonClassNames } from './Button.Props';
 import * as stylesImport from './BaseButton.scss';
 const styles: any = stylesImport;
 
-export interface IButtonClassNames {
-  base?: string;
-  variant?: string;
-  isDisabled?: string;
-  isEnabled?: string;
-  description?: string;
-  flexContainer?: string;
-  icon?: string;
-  menuIcon?: string;
-  label?: string;
-  root?: string;
-}
-
-/**
- * These props are not in the Props file as they are undocumented props only specific to BaseButton.
- *
- * @export
- * @interface IBaseButtonProps
- * @extends {IButtonProps}
- */
-export interface IBaseButtonProps extends IButtonProps {
-  /**
-   *  Custom class names for individual elements within the button DOM.
-   */
-  classNames?: IButtonClassNames;
-}
-
 export interface IBaseButtonState {
   menuProps?: IContextualMenuProps | null;
 }
 
-export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState> implements IButton {
+export class BaseButton extends BaseComponent<IButtonProps, IBaseButtonState> implements IButton {
 
-  public static defaultProps: IBaseButtonProps = {
+  public static defaultProps: IButtonProps = {
```

Comment:
I just realized this is a really bad contract; for example, it is unclear from a user how to define the color on an icon that is disabled. We need to tweak this a bit.

Can we hold off adding something that would be deprecated fast just for a sec? Sorry, know this has been a long time.